go/vt/vtctl/grpcvtctldserver/server.go

```diff
@@ -663,49 +700,85 @@ func (s *VtctldServer) GetTablets(ctx context.Context, req *vtctldatapb.GetTable
 		cells = c
 	}
 
-	var allTablets []*topodatapb.Tablet
+	var (
+		m          sync.Mutex
+		wg         sync.WaitGroup
+		rec        concurrency.AllErrorRecorder
+		allTablets []*topo.TabletInfo
+	)
 
 	for _, cell := range cells {
-		tablets, err := topotools.GetAllTablets(ctx, s.ts, cell)
-		if err != nil {
-			return nil, err
-		}
-
-		// Collect true master term start times, and optionally filter out any
-		// tablets by keyspace according to the request.
-		masterTermStartTimes := map[string]time.Time{}
-		filteredTablets := make([]*topo.TabletInfo, 0, len(tablets))
+		wg.Add(1)
 
-		for _, tablet := range tablets {
-			if req.Keyspace != "" && tablet.Keyspace != req.Keyspace {
-				continue
-			}
+		go func(cell string) {
+			defer wg.Done()
 
-			key := tablet.Keyspace + "." + tablet.Shard
-			if v, ok := masterTermStartTimes[key]; ok {
-				if tablet.GetMasterTermStartTime().After(v) {
-					masterTermStartTimes[key] = tablet.GetMasterTermStartTime()
+			tablets, err := topotools.GetAllTablets(ctx, s.ts, cell)
+			if err != nil {
+				if req.Strict {
+					log.Infof("GetTablets got an error from cell %s: %s. Running in strict mode, so canceling other cell RPCs", cell, err)
+					cancel()
 				}
-			} else {
-				masterTermStartTimes[key] = tablet.GetMasterTermStartTime()
+
+				rec.RecordError(fmt.Errorf("GetAllTablets(cell = %s) failed: %w", cell, err))
+
+				return
 			}
 
-			filteredTablets = append(filteredTablets, tablet)
+			m.Lock()
+			defer m.Unlock()
+			allTablets = append(allTablets, tablets...)
+		}(cell)
+	}
+
+	wg.Wait()
+
+	if rec.HasErrors() {
+		if req.Strict {
```

Comment:
Super minor point that you are welcome to ignore -- my $0.02 is that `if req.Strict || len(rec.Errors) == len(cells)` is slightly more readable (or at least, what I'd expect, since it took me a sec to realize the two ifs are doing the same thing).

Reply:
Not exactly :) (I ran into this issue when a test broke)

If `len(cells) == 0`, then `len(rec.Errors) == len(cells)` but `rec.HasErrors() == false`, which is why we have to be a little more verbose/redundant here

Reply:
Oh ha got it, thanks

Reply:
I misunderstood! I thought you were suggesting to replace the outer `if rec.HasErrors()` with the composite boolean, not to replace the two branches. You're totally right, and it's way easier to understand. Pushing up a fix! 😊